environment/make.bat

```diff
@@ -1,6 +1,40 @@
-mkdir obj
+@echo off
+setlocal EnableExtensions EnableDelayedExpansion
 
-CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64
+if defined VisualStudioVersion (
+rem vcvarsall has been called already, don't need to do anything ourselves
+) else (
+set vcvarsall_location_1="%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat"
+set vcvarsall_location_2="%ProgramFiles%\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat"
+set vcvarsall_location_3="%ProgramFiles(x86)%\Microsoft Visual Studio 14.0\VC\vcvarsall.bat"
+set vcvarsall_location_4="%ProgramFiles%\Microsoft Visual Studio 14.0\VC\vcvarsall.bat"
+set vcvarsall_location_count=4
+
+for /L %%i in (!vcvarsall_location_count!, -1, 1) do (
+    set vcvarsall_location_candidate=!vcvarsall_location_%%i!
+    if exist !vcvarsall_location_candidate! set vcvarsall_location=!vcvarsall_location_candidate!
+)
+
+if not defined vcvarsall_location (
+    echo Failed to find vcvarsall.bat in any of the known places. You have two options:
+    echo 1^) Preferred: run vcvarsall.bat yourself before running this script. Check out https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line for more information.
+    echo 2^) Find where vcvarsall.bat file is on your system and add it to the list of locations in this batch file.
+    pause
+    exit /b 1
+)
+
+reg query "HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" /v PROCESSOR_ARCHITECTURE | find /i "x86" > nul
+if !ERRORLEVEL! == 0 (
```

Comment:
I'm guessing that the systems used to build the 32bit binaries currently are actually 64bit systems. So there probably needs to be a way to manually override this.

Reply:
You'd run vcvarsall.bat yourself before calling this script, which is how MS intended you use this.

Reply:
Also we don't actually build any 32 bit binaries anywhere ourselves. Some people asked for being able to build on 32 bit windows, since that's the only thing they had. On those machines it should "just work".

Reply:
Oops, sorry I thought I had seen a 32bit download for windows before. Must have just been the .bat file.